gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemBase.java

```diff
@@ -1454,14 +1487,22 @@ private GoogleCloudStorageFileSystem createGcsFs(Configuration config) throws IO
     GoogleCloudStorageFileSystemOptions gcsFsOptions =
         GoogleHadoopFileSystemConfiguration.getGcsFsOptionsBuilder(config).build();
 
+    checkArgument(gcsFsOptions.getCloudStorageOptions().getAuthorizationMode()
+            != AuthorizationMode.REQUEST_CONTEXT_RELATED
+            || (HadoopCredentialConfiguration.getNullCredentialEnabled(config, GCS_CONFIG_PREFIX)
+            && HadoopCredentialConfiguration.getServiceAccountEnabled(config, GCS_CONFIG_PREFIX)),
+        "When using AuthorizationMode==REQUEST_CONTEXT_RELATED, `fs.gs.auth.null.enabled` should be true and `fs.gs.auth.service.account.enable` should be false");
```

Comment:
Can we move this condition in the `getAccessTokenProvider` provider, so all `AuthorizationMode` requests are in the same place?

Reply:
Of course.